packages/eas-cli/src/commands/env/init.ts

```
@@ -0,0 +1,231 @@
+    } = await this.parse(EnvironmentVariableInit);
+
+    if (nonInteractive) {
+      throw new Error("Non-interactive mode is not supported for 'eas env:init'");
+    }
+
+    const { projectDir } = await this.getContextAsync(EnvironmentVariableInit, {
+      nonInteractive,
+    });
+
+    await this.ensureDirenvInstalledAsync(projectDir);
+    await this.setupEnvrcFileAsync(projectDir);
+    await this.addDirenvHookToShellConfigAsync();
+    await this.addToGitIgnoreAsync(projectDir);
+    Log.log('Running direnv allow...');
+    await spawnAsync('direnv', ['allow'], { cwd: projectDir, stdio: 'inherit' });
+  }
+
+  private async addDirenvHookToShellConfigAsync(): Promise<void> {
+    const direnvConfig = this.getShellDirenvConfig();
+
+    if (direnvConfig && (await pathExists(direnvConfig.shellConfigPath))) {
+      const { shellConfigPath, direnvHookCmd, direnvInitCmd } = direnvConfig;
+
+      const confirm = await confirmAsync({
+        message: `Do you want to add the direnv hook to ${shellConfigPath}?`,
+      });
+
+      if (!confirm) {
+        Log.log('Skipping adding the direnv hook to the shell config');
+        Log.log('You may need to add the direnv hook to your shell config manually.');
+        Log.log('Learn more: https://direnv.net/docs/hook.html');
+        return;
+      }
+
+      const configContent = await readFile(shellConfigPath, 'utf8');
+
+      if (configContent.includes(direnvHookCmd)) {
+        Log.log('The direnv hook is already present in the shell config');
+        return;
+      }
+
+      await appendFile(shellConfigPath, `\n${direnvHookCmd}\n`, 'utf8');
+      Log.log(`Added direnv hook to ${shellConfigPath}`);
+      await spawnAsync(...direnvInitCmd);
+    } else {
+      Log.log("Unable to determine the user's shell");
+      Log.log('You may need to add the direnv hook to your shell config manually.');
+      Log.log('Learn more: https://direnv.net/docs/hook.html');
+    }
```

I believe we also need to throw an error here